homeassistant/components/gpsd/sensor.py

```diff
@@ -77,7 +77,13 @@ def setup_platform(
 class GpsdSensor(SensorEntity):
     """Representation of a GPS receiver available via GPSD."""
 
-    def __init__(self, hass, name, host, port):
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        name: str | None = None,
```

Comment:
Name, host and port have a default value, so they can't be None.

Reply:
If I remove None, I get:

~~~
% mypy homeassistant/components/gpsd           
homeassistant/components/gpsd/sensor.py:74: error: Argument 2 to "GpsdSensor" has incompatible type "Any | None"; expected "str"  [arg-type]
homeassistant/components/gpsd/sensor.py:74: error: Argument 3 to "GpsdSensor" has incompatible type "Any | None"; expected "str"  [arg-type]
homeassistant/components/gpsd/sensor.py:74: error: Argument 4 to "GpsdSensor" has incompatible type "Any | None"; expected "int"  [arg-type]
Found 3 errors in 1 file (checked 2 source files)
~~~

Reply:
Maybe change the .get() to [CONF_NAME] above

Reply:
Thanks :)